src/System.Private.CoreLib/src/System/Environment.cs

```diff
@@ -50,6 +50,20 @@ internal static long TickCount64
             }
         }
 
+        public static string MachineName
+        {
+            get
+            {
+                const int MaxMachineNameLength = 256;
+                StringBuilder buf = new StringBuilder(MaxMachineNameLength);
+                int capacity = MaxMachineNameLength;
+                if (Interop.mincore.gethostname(buf, capacity) != 0)
```

Comment:
`gethostname` requires winsock to be initialized. I have tried a simple Console.WriteLine(Environment.MachineName) using your implementation. It returned failed with invalid operation exception - because of the call to gethostname returned error "Either the application has not called WSAStartup, or WSAStartup failed.".